packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -556,17 +556,16 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         emit TransactionDeposited(from, _to, DEPOSIT_VERSION, opaqueData);
     }
 
-    /// @notice Sets the gas paying token for the L2 system. This token is used as the
-    ///         L2 native asset. Only the SystemConfig contract can call this function.
-    function setGasPayingToken(address _token, uint8 _decimals, bytes32 _name, bytes32 _symbol) external {
+    /// @notice Sets static configuration options for the L2 system. Only the SystemConfig contract can call this
+    ///         function.
+    function setConfig(ConfigType _type, bytes memory _value) external {
```

Comment:
We don't want to modify the `setGasPayingToken` implementation. We should add a new path that is generic and then be able to set the gas paying token that way and also be able to set other static attributes such as dep set members or batcher hash. Then we can remove this path in the future

Reply:
good idea!